password_security/models/res_users.py

```diff
@@ -185,18 +185,20 @@ def _check_password_history(self, password):
         :raises: UserError on reused password
         """
         crypt = self._crypt_context()
-        for rec_id in self:
-            recent_passes = rec_id.company_id.password_history
-            if recent_passes < 0:
-                recent_passes = rec_id.password_history_ids
+        for user in self:
+            password_history = user.company_id.password_history
+            if not password_history:  # disabled
+                recent_passes = self.env["res.users.pass.history"]
```

Comment:
I personally prefer to see .browse() after the model reference, as I think it's a more obviously understandable idiom, but I understand that they are synonyms.

```
recent_passes = self.env["res.users.pass.history"].browse()
```